spec_cleaner/rpmpreamble.py

```diff
@@ -445,12 +482,16 @@ def add(self, line):
             self._add_line_value_to('patternprovides', match.group(1), key='Provides')
             return
 
-        elif self.reg.re_provides.match(line) and self.reg.re_patternobsolete.search(line):
+        elif self.reg.re_provides.match(line) and self.reg.re_patternobsolete.search(
+            line
+        ):
```

Comment:
the one line here looked way better, figure way to make black format it in better way.

Reply:
Black doesn't have many configuration options intentionally, it just follows what pep8 says. In this case, we can only extend the max-line-length from 88 to something higher (but it can complicate e.g. two columns views for editors/code reviews/patches etc.), or mark specific lines with "# fmt" tag which says black no to touch it. In some cases, it simply helps to have pep8 in your mind during coding and limit all lines to a maximum of 88 (or another number) characters so black doesn't have to touch it.

Reply:
In most of my projects (and other rpm projects), I've got black set to 100 characters for line length. That seems to work well enough for coding and reviewing.

Reply:
Yea or the newline can be inserted sooner, basically just having that one variable only looks weird tbh.

Reply:
Yes, that single variable on a separate line looks weird indeed. But if I try to insert the newline sooner, black always reformats it in the same way :/

Reply:
longer line limit?

Reply:
I tried to set the line limit to 100 chars as we discussed. Obviously it helped that mentioned lines but it created another with the same "problem":

```diff
-            line = line.replace(
-                '%{' + i + '_version}', '0%{?' + i + '_version}'
-            ).replace('00%{', '0%{')
+            line = line.replace('%{' + i + '_version}', '0%{?' + i + '_version}').replace(
+                '00%{', '0%{'
+            )
```

We can't completely avoid it (just to write shorter lines by the default). So should I keep it on 88 chars or 100 chars?

Reply:
Let's keep at 100. It minimizes it and accounts for modern screen width.

Reply:
done